radsat-sk/operation/subsystems/RAntenna.c

```diff
@@ -0,0 +1,293 @@
+/**
+ * @file RAntenna.c
+ * @date Jan 15, 2022
+ * @author Addi Amaya (caa746)
+ */
+
+#include <RAntenna.h>
+#include <satellite-subsystems/IsisAntS.h>
+#include <string.h>
+#include <hal/errors.h>
+#include <hal/Timing/Time.h>
+
+/***************************************************************************************************
+                                  PRIVATE DEFINITIONS AND VARIABLES
+***************************************************************************************************/
+
+/** Struct for defining Antenna Systems I2C Addresses. Configured as a single bus, both address write to same location */
+ISISantsI2Caddress RAntennaI2Caddress = {ANTENNA_I2C_SLAVE_ADDR_PRIMARY,ANTENNA_I2C_SLAVE_ADDR_REDUNANT};
+
+/** Struct that holds the current deployment status of the antenna*/
+antenna_deployment_status_t RAntennaStatus;
+
+/** Track whether the antenna has been initialized yet */
+static int antennaInitialized = 0;
+
+/** Track Number of times the antenna attempted to deploy*/
+static int antennaDeploymentAttempts = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Initializes the ISISpace Antenna driver
+ *
+ * @return 0 for success, non-zero for failure. See hal/errors.h for details.
+ */
+int antennaInit(void) {
```

Comment:
I think the deployments should be abstracted out into their own private function because we do the exact same thing on both sides.

Reply:
I believe this comment is saying to make something like

`init antennaInit(void) {
       initAside();
       initBSide();
}
`
I am not sure how you would go about implementing this for the antenna Init as the ISIS's antenna initialization function works for both sides, as detailed in the code